src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -420,13 +420,17 @@ export class MockHttpSocket extends MockSocket {
   }
 
   private flushWriteBuffer(): void {
-    for (const [, , callback] of this.writeBuffer) {
-      /**
-       * @note If the write callbacks are ever called twice,
-       * we need to mark them with a symbol so they aren't called
-       * again in the `passthrough` method.
-       */
-      callback?.()
+    for (const writeCall of this.writeBuffer) {
+      if (typeof writeCall[2] === 'function') {
+        writeCall[2]()
+        /**
+         * @note Remove the callback from the write call
+         * so it doesn't get called twice on passthrough
+         * if `request.end()` was called within `request.write()`.
+         * @see https://github.com/mswjs/interceptors/issues/684
+         */
+        writeCall[2] = undefined
```

Comment:
I decided to go with the replacement approach because it's simpler than introducing an internal symbol.

Reply:
I agree.